docs/source/howto/codes.rst

```diff
@@ -4,41 +4,264 @@
+The ``code_uuid`` being passed in line 13 is necessary for the engine to get the base information from the |Code| node (in which computer the code is placed, what is the location of the executable, etc).
+Note that this was taken in the unloading block from ``input_code = self.inputs['code']``, which we never specified in the ``define`` method: this is one of the inputs defined in the base |CalcJob| class that we mentioned earlier when discussing the ``super().define(spec)``:
+
+.. code-block:: python
+
+    spec.input('code', valid_type=orm.Code, help='The `Code` to use for this job.')
+
+Moreover, as this information is necessary for the engine, this input has a ``required=True`` setting (the default for all inputs, including the ones we manually defined earlier).
+For the plugin user this means they will have to provide it when instantiating the calculation (as can be seen in the :ref:`respective section<how-to:codes:run>` below), whereas for the plugin developer this means they will have to make sure to manually pass its UUID from the ``self.inputs['code']`` node to the ``codeinfo.code_uuid`` property, as shown here.
+
+The other two lines are configuring how to build the running script: line 14 indicates where to redirect the standard output, whereas line 15 lists the command line arguments to be passed to the code.
+The specific combination presented here, together with the information inside of the ``code`` node, results in the following script:
+
+.. code-block:: bash
+
+    #!/bin/bash
+
+    '[executable path in code node]' '[input_filename]' > '[output_filename]'
+
+Through the |CodeInfo| object you can also pass more flags (by adding them as string elements to the list in ``codeinfo.cmdline_params``), configure what to pass through the standard input (just as it is shown for the standard output), add commands to be run before and after the execution line, etc.
+
+Finally, the last block remaining in lines 17-19 creates the |CalcInfo| object, passes to it the |CodeInfo| object, and adds the output to the ``retrieve_list``.
+This is a list of all files that the engine needs to copy back from the computer where the code is located, either for immediate parsing or local storage in an output node labeled ``retrieved`` for future post-processing (again, this is another base ``output`` defined in the base |CalcJob| class).
+You can expand this list to include more files, and there are other lists available that allow you to easily specify which files to make available in the remote folder without having to manually copy them during this step and thus having to store unnecessary copies of them in the calculation node.
+
+In the Topics section on :ref:`defining calculations <topics:calculations:usage>` you will find more information on available settings of the |CalcInfo| and |CodeInfo|, such as available copy lists, running script options, etc.
 
 Design guidelines
-------------------
+-----------------
```

Comment:
Something is off here. Why is this followed by the "Design guidelines" completely out of context? Did something go wrong during a merge/rebase?

Reply:
"Design guidelines" was something that @ltalirz  put here in one of his recen PR's, I think the plugin ones. I was not sure how to incorporate them into the text of the how-to so I left it out as a final sub-section (I only separated the item that referred to parsing and moved it at the end of the respective section). I'm not really sure what to do with it other than this but I also didn't want to just delete it.

Reply:
Ok, then let's just leave it there and I'll try to figure it out later.